### PR TITLE
Some changes in Item/block translating

### PR DESCRIPTION
**Pull request by Creeperface01**

### CODE REVIEW

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCMultiBlockChangePacketTranslator.java [hunk 1]
```diff
@@ -37,10 +39,15 @@ public PCMultiBlockChangePacketTranslator() {
...
+
+			BlockState block = packet.getRecords()[i].getBlock();
+			ItemEntry entry = ItemBlockTranslator.translateToPE(block.getId(), block.getData());
+
+			packets[i].id = (byte) (entry.id
 					& 0xFF);
-			packets[i].data = generalFlag;
-			packets[i].flags = packet.getRecords()[i].getBlock().getData();
+			packets[i].flags = generalFlag;
+			packets[i].data = entry.damage;
```
Comment (Creeperface01) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCMultiBlockChangePacketTranslator.java:50]: That was also wrong I think

**Comment by HoverEpic** [comment]
Sorry, i just commit ~ the same thing =P

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCMultiChunkDataPacketTranslator.java [hunk 2]
```diff
@@ -72,8 +73,14 @@ private void processChunkSection(Chunk[] pc, ChunkData pe) {
...
-					pe.sections[cy].blockIds[index(x, y,
-							z)] = (byte) (ItemBlockTranslator.translateToPE(block.getId()) & 0xFF);
+					ItemEntry entry = ItemBlockTranslator.translateToPE(block.getId(), block.getData());
+
+					Section section = pe.sections[cy];
+					section.blockIds[index(x, y,
+							z)] = (byte) (entry.id & 0xFF);
+
+					section.blockMetas[index(x, y,
+							z)] = entry.damage.byteValue();
```
Comment (HoverEpic) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCMultiChunkDataPacketTranslator.java:83]: GG, i was on this one, nice work =)
Have you got this code working?

`java.lang.ArrayIndexOutOfBoundsException: 2048
	at org.dragonet.proxy.network.translator.pc.PCMultiChunkDataPacketTranslator.processChunkSection(PCMultiChunkDataPacketTranslator.java:83)
	at org.dragonet.proxy.network.translator.pc.PCMultiChunkDataPacketTranslator.lambda$translate$0(PCMultiChunkDataPacketTranslator.java:46)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)`

**Review by HoverEpic — APPROVED** [review]

**Comment by Creeperface01** [comment]
Still I think we should create comething better than current translation system 😅 for inventory we would need to translate items from pe to PC again

File: proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java [hunk 3]
```diff
@@ -26,35 +26,65 @@
...
+		toPEOverride(126, 158); // Slab <-> NULL
+		toPEOverride(95, 241); // Stained Glass
+		toPEOverride(36, 248);
+		toPEOverride(166, 95); //barrier -> invisible bedrock
+		toPEOverride(188, new ItemEntry(85, 1));
+		toPEOverride(189, new ItemEntry(85, 2));
+		toPEOverride(190, new ItemEntry(85, 3));
+		toPEOverride(191, new ItemEntry(85, 5));
+		toPEOverride(192, new ItemEntry(85, 4));
+		toPEOverride(203, 202); //purpur stairs
```
Comment (HoverEpic) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java:44]: add also :
- (157, 28); //detector_rail
- (166, 95); //barrier -> invisiblebedrock
- (168, 248); //prismarine -> update_block
- (169, 248); //prismarine_bricks -> update_block
- (198, 208); //end_rod
- (199, 240); //chorus_plant
- (202, 201); //purpur_pillar -> purpur_block
- (208, 198); //grass_path
- (210, 188); //repeated command block
- (211, 189); //chained command block

**Review by HoverEpic — APPROVED** [review]

File: proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java (hunk 3 above)
Comment (Creeperface01) [reply proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java:44]: I think we don't have to add update blocks. Minecraft automatically replaces unknown blocks with update block

**Pull request merged**